src/rpc_mock.rs

```diff
@@ -0,0 +1,74 @@
+// Implementation of RpcRequestHandler trait for testing Rpc requests without i/o
+
+use crate::rpc_request::{RpcClient, RpcError, RpcRequest, RpcRequestHandler};
+use serde_json::{self, Number, Value};
+use solana_sdk::hash::Hash;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_sdk::system_transaction::SystemTransaction;
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::io::{Error, ErrorKind};
+use std::net::SocketAddr;
+
+pub const PUBKEY: &str = "7RoSF9fUmdphVCpabEoefH81WwrW7orsWonXWqTXkKV8";
+pub const SIGNATURE: &str =
+    "43yNSFC6fYTuPgTNFFhF4axw7AfWxB2BPdurme8yrsWEYwm8299xh8n6TAHjGymiSub1XtyxTNyd9GBfY2hxoBw8";
+
+pub struct MockRpu {}
+
+impl RpcRequestHandler for MockRpu {
+    fn make_rpc_request(
+        client: &RpcClient,
+        _id: u64,
+        request: RpcRequest,
+        params: Option<Value>,
+    ) -> Result<Value, Box<dyn error::Error>> {
+        if &client.addr == "fails" {
+            return Ok(Value::Null);
+        }
+        if request == RpcRequest::ConfirmTransaction {
+            if let Some(Value::Array(param_array)) = params {
+                if let Value::String(param_string) = &param_array[0] {
+                    return Ok(Value::Bool(param_string == SIGNATURE));
+                }
+                Err(RpcError::RpcRequestError("Missing parameter".to_string()))?
+            }
+        } else if request == RpcRequest::GetBalance {
+            if &client.addr == "airdrop" {
```

Comment:
it'd be great if you can get rid of as much duplication as possible. This one, for example, can be written as:

```rust
let n = if &client.addr == "airdrop" { 0 } else { 50 };
Value::Number(Number::from(n))
```